From 9300a87b6866b84ca3ad351a74f4d8d30cf66a5a Mon Sep 17 00:00:00 2001
From: Olivier Brunel <jjk@jjacky.com>
Date: Fri, 9 Jun 2017 22:40:42 +0200
Subject: [PATCH 06/12] quicksearch: expand extended symbols on "Edit"

Instead of silently ignoring the current string when it used extended
symbols, expand it into condition expressions to allow editing it.

--

v2: Fix not working on single-word condition
---
 src/gtk/quicksearch.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/gtk/quicksearch.c b/src/gtk/quicksearch.c
index 6859f662c..fc7a496ad 100644
--- a/src/gtk/quicksearch.c
+++ b/src/gtk/quicksearch.c
@@ -590,12 +590,16 @@ static gboolean search_condition_expr(GtkMenuItem *widget, gpointer data)
 			mainwindow_get_mainwindow()->summaryview->quicksearch != NULL,
 			FALSE);
 
-	/* re-use the current quicksearch value if it's a condition expression,
-	   otherwise ignore it silently */
+	/* re-use the current quicksearch value, expanding it so it also works
+	 * with extended symbols */
 	cond_str = quicksearch_get_text(mainwindow_get_mainwindow()->summaryview->quicksearch);
 
 	if (*cond_str != '\0') {
-		matchers = matcher_parser_get_cond((gchar*)cond_str, NULL);
+		gchar *newstr = advsearch_expand_search_string(cond_str);
+
+		if (newstr && newstr[0] != '\0')
+			matchers = matcher_parser_get_cond(newstr, FALSE);
+		g_free(newstr);
 	}
 
 	prefs_matcher_open(matchers, search_condition_expr_done);
-- 
2.15.1

